### PR TITLE
Add CI for trailing whitespace

### LINKED ISSUE #461: CI checks for typos and blank lines

### PR DESCRIPTION
**Pull request by pxeger**
Closes #461

### CODE REVIEW

**Comment by pxeger** [comment]
pre-commit.ci autofix

**Comment by pxeger** [comment]
cba to deal with this spell checker

**Review by chunkybanana — APPROVED** [review]

File: .github/workflows/linter.yaml [hunk 1]
```diff
@@ -7,16 +7,14 @@ on:
   push:
     branches: [ main ]
   pull_request:
-    branches: [ main ]
+    branches-ignore: [ production ]
```
Comment (cgccuser) [inline .github/workflows/linter.yaml:10]: Don't we want to run tests on prod?

**Review by cgccuser — APPROVED** [review]
Looks good overall

File: .github/workflows/linter.yaml (hunk 1 above)
Comment (pxeger) [reply .github/workflows/linter.yaml:10]: but everything on prod should have come from main, so we don't need it to run again

**Pull request merged**